File: go/int-ai4.go
package main

import (
    "fmt"
    "math"
)

// TrapezoidalRule computes the integral of a function f between a and b using n intervals
func TrapezoidalRule(f func(float64) float64, a, b float64, n int) float64 {
    dx := (b - a) / float64(n)
    sum := 0.0
    x := a
    for i := 0; i < n; i++ {
        sum += (f(x) + f(x+dx)) / 2 * dx
        x += dx
    }
    return sum
}

// AbsSin is the function |sin(x)|
func AbsSin(x float64) float64 {
    return math.Abs(math.Sin(x))
}

func main() {
    // Integration limits
    a := -math.Pi
    b := 2.0 / 3.0 * math.Pi
    n := 10000  // Number of intervals

    // Compute the area using the Trapezoidal Rule
    area := TrapezoidalRule(AbsSin, a, b, n)

    // Print the computed area
    fmt.Printf("The total area between sin(x) and the x-axis from -pi to 2/3*pi is: %f\n", area)
}

File: julia/int-ai4.jl
# Function to implement the trapezoidal rule for integration
function trapezoidal_rule(f, a, b, n)
    dx = (b - a) / n
    area = 0.0
    for i = 1:n
        x_i = a + (i-1) * dx
        x_next = a + i * dx
        area += 0.5 * (f(x_i) + f(x_next)) * dx
    end
    return area
end

# Function for the absolute value of sin(x)
abs_sin(x) = abs(sin(x))

# Define the limits of integration and number of intervals
a = -pi
b = 2/3 * pi
n = 10000  # Using a large number of intervals for better accuracy

# Compute the area using the trapezoidal rule
area = trapezoidal_rule(abs_sin, a, b, n)

# Output the computed area
println("The total area between sin(x) and the x-axis from -pi to 2/3*pi is: ", area)
